app.js

```diff
@@ -0,0 +1,52 @@
+//Buttons Selectors
+const incrementBtn = document.querySelectorAll('.plus');
+const decrementBtn= document.querySelectorAll('.minus');
+const favBtn = document.querySelectorAll('.favorite');
+const deleteBtn = document.querySelectorAll('.remove');
+
+//Chnaging Values Selectors
+const productQuantity = document.querySelectorAll('.number');
+const subtotalAmount = document.querySelector('.subtotal-amount');
+const totalAmount = document.querySelector('.total-amount');
+const delivery = document.querySelector('.shipping-amount').innerText.slice(1);
+const pricesArr = document.querySelectorAll('.cart-item-price');
+let total = 0;
+//logic for the remove button
+for (let i = 0; i < deleteBtn.length; i++){
+    deleteBtn[i].addEventListener("click", (e)=>{
+        let btnClicked = e.target;
+        btnClicked.parentElement.remove();
+    });
+}
+//logic for favorite button
+for(let i = 0; i < favBtn.length; i++){
+    favBtn[i].addEventListener("click", (e)=>{
+        let fav = e.target;
+        fav.classList.toggle('favorite-active');
+    });
+}
+//parsing delivery amount into number
+
+//increment button function 
+for (let i = 0; i < incrementBtn.length; i++){
+    incrementBtn[i].addEventListener("click", (e)=>{
+        const quantity = e.target.nextElementSibling;
+        quantity. innerText = Number( e.target.nextElementSibling.innerText ) + 1;
+        total = total + (Number(pricesArr[i].innerText.slice(1)));
```

Comment:
I think it's better to use a custom attribute in the HTML and put the price of each item rather than depending on the text displayed to the user in the HTML because it's not a very reliable source.

This can be achieved in the HTML by using the custom `data` attribute like this:

```html
<div class="cart-item-price" data-price="30">$30.00</div>
```

and access this like in your JS code:

```js
const priceElements = document.querySelectorAll('[data-price]');
```

and inside your event listener in the loop:

```js
const price = Number(priceElements[i].dataset.price);
total += price;
```